### PR TITLE
Add initial http server stuff

### PR DESCRIPTION
**Pull request by OFFTKP**
PANDA? PANDA.
(repeat 31 times)

### CODE REVIEW

**Comment by OFFTKP** [comment]
lmk what you think, I'll add another commit soon

**Comment by OFFTKP** [comment]
whoopsie

**Comment by wheremyfoodat** [comment]
This is starting to get too big and should be moved to a dedicated file instead of emulator.cpp

**Comment by OFFTKP** [comment]
> This is starting to get too big and should be moved to a dedicated file instead of emulator.cpp

I dreaded this comment

Let's see if it compiles fine and I'll move it in an upcoming commit

**Comment by Wunkolo** [comment]
Also I'd rather `Emulator::screenshot` just called a member-function to the rendering backend than use OpenGL directly.

**Comment by wheremyfoodat** [comment]
Good idea, agreed

File: src/httpserver.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+		{"Left", Keys::Left},
+		{"Up", Keys::Up},
+		{"Down", Keys::Down},
+		{"R", Keys::R},
+		{"L", Keys::L},
+		{"X", Keys::X},
+		{"Y", Keys::Y},
+		{"CirclePadRight", Keys::CirclePadRight},
+		{"CirclePadLeft", Keys::CirclePadLeft},
+		{"CirclePadUp", Keys::CirclePadUp},
```
Comment (skylersaleh) [inline src/httpserver.cpp:30]: The SkyEmu protocol supports floating point values as inputs, so maybe CirclePadX and CirclePadY might make more sense than, up/down/left/right.

Comment (OFFTKP) [reply src/httpserver.cpp:30]: I'll change it in upcoming patch

**Review by wheremyfoodat — APPROVED** [review]

**Comment by georgemoralis** [comment]
panda!

**Comment by wheremyfoodat** [comment]
panda?

**Comment by georgemoralis** [comment]
ΜΟΝΟ ΠΑΝΤΑ ΤΡΕΛΑ ΚΑΙ ΚΟΡΔΕΛΑ

**Pull request merged**